src/Provider/AbstractProvider.php

```diff
@@ -674,6 +674,10 @@ protected function parseResponse(ResponseInterface $response)
                 throw $e;
             }
 
+            if ($response->getStatusCode() == 500) {
+                throw new UnexpectedValueException('Failed to parse JSON and OAuth server response code is 500. It could be internal error on Oauth server.');
```

Comment:
Code style also needs to be fixed here, as this newly added line goes way over 120 characters.